turbopack/crates/turbo-tasks-backend/src/kv_backing_storage.rs

```diff
@@ -449,6 +489,32 @@ impl<T: KeyValueDatabase + Send + Sync + 'static> BackingStorage
     }
 }
 
+fn restore_strings<D: KeyValueDatabase>(
+    database: &D,
+    tx: &D::ReadTransaction<'_>,
+    global_ids: &LocalIdToGlobalId,
+) -> Result<LocalIdToRcStr> {
+    let mut map = Vec::with_capacity(global_ids.len());
+
+    for global_id in global_ids.iter() {
+        let Some(value) = database.get(
+            tx,
+            KeySpace::ReverseStringInternMap,
+            IntKey::new(global_id).as_ref(),
+        )?
+        else {
+            bail!("Unable to find string for {global_id}")
+        };
+
+        map.push(unsafe {
+            // Safety: We interned a rust string, so it is valid utf-8
+            RcStr::from(str::from_utf8_unchecked(value.borrow()))
+        });
```

Comment:
We need some kind of cache here. Otherwise it will query the DB for every string in every value, which will multiply db load by a lot. And also it won't deduplicate RcStrs in memory after deserialization.

But we could do that as follow-up PR if we accept a temporary regression of Persistent Caching